keymanager-lib/src/methods.rs

```diff
@@ -30,6 +39,35 @@ pub fn get_public_key(req: &RequestIds, _ctx: &mut RpcContext) -> Result<Option<
     pk.map_or(Ok(None), |pk| Ok(Some(kdf.sign_public_key(pk)?)))
 }
 
+/// See `Kdf::get_or_create_keys`.
+pub fn get_or_create_ephemeral_keys(
+    req: &EphemeralKeyRequest,
+    ctx: &mut RpcContext,
+) -> Result<KeyPair> {
+    // Derive ephemeral keys only for epochs which are not too far back in the past.
```

Comment:
Don't also forget about client authentication (possibly refactor a bit to avoid too much duplication) here as otherwise anyone could derive private ephemeral keys.